gameEditor/gameEditor.js

```diff
@@ -0,0 +1,174 @@
+            let child_prompt = options[option]["option"];
+            children.push({prompt: child_prompt, node: child_node});
+            child_node.parents.push(this.nodes[current_num]);
+        }
+
+        
+        this.nodes[current_num].children = children;
+        return this.nodes[current_num];
+    }
+
+    createNode(num, text) {
+        this.nodes[num] = new Node(text, num);
+    }
+
+    linkNodes(parent, child, prompt) {
+        parent.children.push((child, prompt));
+    }
+
+    deleteNode(node) {
+        for (var index in node.parents) {
+            parent = node.parents[index][0];
+            delete parent[node.parents[index][1]];
+        }
+        delete this.nodes.map(e => e.text).indexOf(node.text);
+    }
+}
+
+export function editGameFile() {
+
+    const fileInput = document.getElementById("game-file");
+    const file = fileInput.files[0];
+
+    const fileDiv = document.getElementById("file-div");
+    fileDiv.parentNode.removeChild(fileDiv);
+
+    var reader = new FileReader();
+    reader.readAsText(file);
+
+    reader.onload = readerEvent => {
+        var gameData = readerEvent.target.result;
+        createGameEditor(JSON.parse(gameData));
+        visualize(editor.root);
+    }
+
+}
+
+function createGameEditor(gameData) {
+    editor = new GameEditor(gameData);
+}
+
+
+function visualize(root) {
+    let visited = new Set();
+    d3.select("#teal-ui").append("svg").attr("viewBox", "0 0 2000 2000").attr("id", "editor-canvas");
+
+    var rootSvg = visualizeNode(root);
+    visited.add(root.num, "50");
+    var dpOneSvg = [];
+    var dpTwoSvg = [];
+
+    
+    for (let keyOne in root.children) {
+        let child = root.children[keyOne]["node"];
+        if (visited.has(child.num)) {
+            continue;
+        }
+
+        dpOneSvg.push(visualizeNode(child, "150"));
+        for(let keyTwo in child.children) {
+            if (visited.has(child.children[keyTwo]["node"].num)) {
+                continue;
+            }
+
+            dpTwoSvg.push(visualizeNode(child.children[keyTwo]["node"], "300"));
+            visited.add(child.children[keyTwo]["node"].num);
+        }
+    }
+
+    // translate all created nodes to correct position and add arrows
+    translateNodes(rootSvg, dpOneSvg, dpTwoSvg);
+    
+}
+
+function visualizeNode(node, yVal) {
+    var g = d3.select("#editor-canvas").append("g").attr("id", node.num);
+    var svg = g.append("svg").attr("width" , "200").attr("height", "60").attr("y", yVal);
+    svg.append('rect')
+        .attr("fill", "black")
+        .attr('stroke', 'black')
+        .attr('width', 200)
+        .attr('height', 50)
+        .attr("rx", "10%")
+        .attr("ry", "10%")
+
+    svg.append('text')
+        .attr("transform", "translate(90,30)")
+        .attr("fill", "#00a6fb")
+        .text(node.num);
+
+    return g;
+}
+
+
+function translateNodes(rootSvg, dpOneSvg, dpTwoSvg) {
```

Comment:
Seems like we are calculating node placement based on depth here. Please add comments that describe what the code does here, and elsewhere.